### PR TITLE
_init_sack() (fix #74)

### LINKED ISSUE #74: implement _init_sack() method in Base class that would init dnf sack
https://github.com/timlau/dnf-apiex/blob/master/dnf-install.py - thats how dnf sack is initiated from it's API

### PR DESCRIPTION
**Pull request by regeciovad**

### CODE REVIEW

File: rpg.spec [hunk 1]
```diff
@@ -7,6 +7,8 @@ URL:            https://github.com/rh-lab-q/rpg
 Source:         %{name}-%{version}.tar.gz
 BuildRequires:  cmake
 BuildRequires:  python3-devel
+BuildRequires:  dnf
+BuildRequires:  python3-dnf
```
Comment (cibo94) [inline rpg.spec]: Does build really requires dnf and python3-dnf?

Comment (jsilhan) [inline rpg.spec]: Good note. If it's a build requirement or needed dependency for running tests then should be there. We don't need it for `make test` to run properly. We should always mock that otehrwise DNF would load metadata which takes a lot of time. Delete these lines, please.

File: rpg.spec [hunk 2]
```diff
@@ -16,6 +18,8 @@ Requires:       coreutils
 Requires:       file
 Requires:       makedepend
 Requires:       rpmdevtools
+Requires:       dnf
```
Comment (jsilhan) [inline rpg.spec]: Actually only `python3-dnf` is needed

File: rpg/__init__.py [hunk 3]
```diff
@@ -24,11 +26,18 @@ def __init__(self):
...
         self._plugin_engine = PluginEngine(self.spec, self.sack)
         self._source_loader = SourceLoader()
         self._copr_uploader = CoprUploader()
+        
+    def _init_sack(self):
+        with dnf.Base() as self._dnf_base:
+            self._dnf_base.conf.releasever = dnf.rpm.detect_releasever(self._dnf_base.conf.installroot)
+            self._dnf_base.read_all_repos()
+            self._dnf_base.fill_sack()
+            return self._dnf_base
```
Comment (jsilhan) [inline rpg/__init__.py]: return self._dnf_base.sack

**Comment by jsilhan** [comment]
all merged. Good job. Thanks

**Pull request closed without merging**